src/compile/inference/top_level_definitions.pegjs

```diff
@@ -0,0 +1,333 @@
+SourceUnit =
+    tlds: (__ t: TopLevelDefinition __  { return t; })* {
+        // Dummy uses to silence unused function TSC errors in auto-generated code
+        expected;
+        error;
+        return tlds as TopLevelNode<any>[];
+    }
+
+TopLevelDefinition
```

Comment:
I guess `FileLevelDefinition` would sound better.

Reply:
I think the term is better, but it requires renaming a lot of things so I got lazy. Lets fix it separately later.